### PR TITLE
Rename $encoding.{Request,Response} to {Req,Res}Meta

### PR DESCRIPTION
**Pull request by abhinav**
Resolves https://github.com/yarpc/yarpc-go/issues/173

Related thriftrw-go change: https://github.com/thriftrw/thriftrw-go/pull/150

@yarpc/golang

### CODE REVIEW

**Comment by breerly** [comment]
ty sir :shipit:

**Comment by breerly** [comment]
```
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:87: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:104: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:121: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:138: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:155: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:172: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:189: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:206: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:223: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:240: response.ResMeta undefined (type thrift.Response has no field or method ResMeta)
crossdock/thrift/gauntlet/yarpc/thrifttestserver/server.go:240: too many errors
FAIL	github.com/yarpc/yarpc-go/crossdock [build failed]
```

File: crossdock/client/gauntlet/behavior.go [hunk 1]
```diff
@@ -379,7 +379,7 @@ func Run(t crossdock.T) {
 		}
 
 		ctx, _ := context.WithTimeout(context.Background(), time.Second)
-		req := thrift.Request{
+		req := thrift.ReqMeta{
```
Comment (prashantv) [inline crossdock/client/gauntlet/behavior.go]: nit: `reqMeta` to match the other renames? (i'm also fine with not renaming, up to you).

**Comment by prashantv** [comment]
lgtm

File: crossdock/client/gauntlet/behavior.go (hunk 1 above)
Comment (abhinav) [inline crossdock/client/gauntlet/behavior.go]: Renamed this and all other references I could find with simple greps.

**Comment by breerly** [comment]
🐑  ittttttt!

**Pull request merged**